### ISSUE TITLE
0.47 checks missing parens for if expression, while 0.46 doesn't (regression?)

### ISSUE THREAD

**Issue by sanyarnd** [pinterest/ktlint#1604]
Maybe it's a false report, but I couldn't find anything related in a changelog.

The following code

```kotlin
fun gcd(a: Int, b: Int): Int {
    return if (b == 0) a
    else gcd(b, a % b)
}
```

now throws
> GcdLcm.kt:4:24: Missing { ... }

The code above passes in 0.46.1

.editorconfig:
```editorconfig
[{*.kt, *.kts}]
ij_kotlin_code_style_defaults = KOTLIN_OFFICIAL

ij_kotlin_imports_layout = *
ij_kotlin_packages_to_use_import_on_demand = ""
ij_kotlin_name_count_to_use_star_import = 999
ij_kotlin_name_count_to_use_star_import_for_members = 999

ij_kotlin_keep_blank_lines_before_right_brace = 1
ij_kotlin_keep_blank_lines_in_code = 1
ij_kotlin_keep_blank_lines_in_declarations = 1

ij_kotlin_assignment_wrap = normal
ij_kotlin_method_call_chain_wrap = normal
ij_kotlin_extends_list_wrap = normal

ij_kotlin_method_parameters_wrap = on_every_item
ij_kotlin_method_parameters_new_line_after_left_paren = true
ij_kotlin_method_parameters_right_paren_on_new_line = true

ij_kotlin_allow_trailing_comma = false
ij_kotlin_allow_trailing_comma_on_call_site = false

ij_kotlin_align_multiline_parameters = false
ij_kotlin_line_comment_at_first_column = false
ij_kotlin_line_break_after_multiline_when_entry = false

```

**Comment by paul-dingemans** [comment]
This is an intended side effect of PR #1565 (https://github.com/pinterest/ktlint/pull/1565/commits/36a9f448aee7e69a9b3eefc25a368321dcc9c817#diff-23102555750e484a7df496237f01c33f59ceca2b0ab415b05de3b05c79c4cd2eL483). It could have been documented more clearly in the changelog.
